php_phongo.c

```diff
@@ -1683,6 +1692,10 @@ void php_phongo_new_regex_from_regex_and_options(zval *object, const char *patte
 	intern->pattern = estrndup(pattern, intern->pattern_len);
 	intern->flags_len = strlen(flags);
 	intern->flags = estrndup(flags, intern->flags_len);
+
+	/* Ensure flags are alphabetized upon initialization. This may be removed
+	 * once CDRIVER-1883 is implemented. */
+	qsort((void *) intern->flags, intern->flags_len, 1, php_phongo_regex_compare_flags);
```

Comment:
I realize this duplicates code from Regex.c; however, it should only be temporary until this is handled in libbson.

I also anticipate refactoring BSON encoding and decoding down the line, which will remove all of the `php_phongo_new_xxx()` functions from this file in favor of using the init functions in the actual BSON class source files.